File: ayurvedaApp/src/main/java/com/cg/medicalapp/service/CustomerService.java
package com.cg.medicalapp.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.cg.medicalapp.entity.Customer;

@Service
public interface CustomerService {

	public Customer addCustomer(Customer customer);
	public Customer removeCustomer(Customer customer);
	public Customer viewCustomer(Customer c);
	public Customer updateCustomer(Customer c);
	public List<Customer> viewAllCustomer();
	
}

File: OAA_Practice/oaa/src/services/CustomerService.js

import axios from 'axios';

const CUSTOMER_API_BASE_URL = "http://localhost:8089/api/v1/customer";
                              
class CustomerService {

    viewAllCustomer(){
        return axios.get("http://localhost:8089/api/v1/customers");
    }

    addCustomer(customer){
        return axios.post(CUSTOMER_API_BASE_URL, customer);
    }

    viewCustomerBycustomerId(customerId){
        return axios.get(CUSTOMER_API_BASE_URL + '/' + customerId);
    }

    updateCustomer(customer){
        return axios.put(CUSTOMER_API_BASE_URL, customer);
    }

    removeCustomer(customerId){
        return axios.delete(CUSTOMER_API_BASE_URL + '/' + customerId);
    }
    
}

export default new CustomerService()